src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -299,7 +300,7 @@ internal unsafe SocketError DoOperationReceive(SafeCloseSocket handle, out Socke
                 socketError = Interop.Winsock.WSARecv(
                     handle,
                     _wsaBufferArray,
-                    _wsaBufferArray.Length,
+                    _bufferListInternal.Count,
```

Comment:
I assume these are changing because we now hold on to _wsaBufferArray and allow it to be larger than the number of items?  If you haven't already, please add comments to these fields describing the relationship.